### PR TITLE
user migration done

### LINKED ISSUE #32: publications migration file

### LINKED ISSUE #33: authors migration file

### LINKED ISSUE #34: users migration file

### PR DESCRIPTION
**Pull request by alamriku**

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -16,9 +16,13 @@ public function up()
         Schema::create('users', function (Blueprint $table) {
             $table->id();
             $table->string('name');
+            $table->string('phone');
             $table->string('email')->unique();
             $table->timestamp('email_verified_at')->nullable();
             $table->string('password');
+            $table->string('role');
+            $table->boolean('is_banned');
```
Comment (me-shaon) [inline database/migrations/2014_10_12_000000_create_users_table.php:24]: use default 'false'.

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 2]
```diff
@@ -16,9 +16,13 @@ public function up()
         Schema::create('users', function (Blueprint $table) {
             $table->id();
             $table->string('name');
+            $table->string('phone');
             $table->string('email')->unique();
             $table->timestamp('email_verified_at')->nullable();
             $table->string('password');
+            $table->string('role');
```
Comment (me-shaon) [inline database/migrations/2014_10_12_000000_create_users_table.php:23]: use default role 'user'. And don't use 'user' as a string here. Declare the Roles in the User Model as constants and use the constant here.

File: database/migrations/2020_10_29_150728_create_authors_table.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('authors', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('image');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_150728_create_authors_table.php:19]: nullable.

File: database/migrations/2020_10_29_152425_create_publications_table.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('publications', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('logo');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_152425_create_publications_table.php:19]: nullable.

File: database/migrations/2020_10_29_153041_create_genres_table.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('genres', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('slug');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_153041_create_genres_table.php:19]: unique.

File: database/migrations/2020_10_29_160742_create_books_table.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('books', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->string('slug');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_160742_create_books_table.php:19]: unique.

File: database/migrations/2020_10_29_160742_create_books_table.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+            $table->string('image');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_160742_create_books_table.php:20]: nullable.

File: database/migrations/2020_10_29_160742_create_books_table.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            $table->unsignedBigInteger('publication_id');
+            $table->string('isbn_number');
+            $table->integer('total_copies');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_160742_create_books_table.php:23]: Default 0.

File: database/migrations/2020_10_29_160742_create_books_table.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            $table->integer('available_copies');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_160742_create_books_table.php:24]: Default 0.

File: database/migrations/2020_10_29_160742_create_books_table.php [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('books');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_160742_create_books_table.php:38]: You need to drop the 'foreign key relationship' first, then drop the table.

Comment (me-shaon) [reply database/migrations/2020_10_29_160742_create_books_table.php:38]: Fix it for other tables too.

File: database/migrations/2020_10_29_163947_create_book_copies_table.php [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('book_copies', function (Blueprint $table) {
+            $table->id();
+            $table->string('uuid');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_163947_create_book_copies_table.php:18]: unique.

File: database/migrations/2020_10_29_163947_create_book_copies_table.php [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+            $table->unsignedBigInteger('book_id');
+            $table->string('edition');
+            $table->string('condition');
+            $table->text('description');
+            $table->dateTime('published_date');
+            $table->boolean('is_available');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_163947_create_book_copies_table.php:24]: Default 'true'.

File: database/migrations/2020_10_29_163947_create_book_copies_table.php [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+            $table->integer('added_by');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_163947_create_book_copies_table.php:25]: You missed the foreign key relationship for this column.

File: database/migrations/2020_10_29_170920_create_loan_requests_table.php [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('loan_requests', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id');
+            $table->string('status');
+            $table->integer('status_changed_by');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_170920_create_loan_requests_table.php:20]: You missed the foreign key relationship for this column.

File: database/migrations/2020_10_29_171427_create_return_requests_table.php [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('return_requests', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id');
+            $table->string('status');
+            $table->integer('status_changed_by');
```
Comment (me-shaon) [inline database/migrations/2020_10_29_171427_create_return_requests_table.php:20]: Add foreign key relationship for this column.

**Review by me-shaon — CHANGES_REQUESTED** [review]

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 1 above)
Comment (alamriku) [reply database/migrations/2014_10_12_000000_create_users_table.php:24]: done

File: database/migrations/2020_10_29_150728_create_authors_table.php (hunk 3 above)
Comment (alamriku) [reply database/migrations/2020_10_29_150728_create_authors_table.php:19]: why authors name should be nullable? image can be nullable undertood that. but name?

Comment (alamriku) [reply database/migrations/2020_10_29_150728_create_authors_table.php:19]: nullable addedd on image

File: database/migrations/2020_10_29_152425_create_publications_table.php (hunk 4 above)
Comment (alamriku) [reply database/migrations/2020_10_29_152425_create_publications_table.php:19]: logo is nullable done

File: database/migrations/2020_10_29_153041_create_genres_table.php (hunk 5 above)
Comment (alamriku) [reply database/migrations/2020_10_29_153041_create_genres_table.php:19]: slug unique done

File: database/migrations/2020_10_29_160742_create_books_table.php (hunk 6 above)
Comment (alamriku) [reply database/migrations/2020_10_29_160742_create_books_table.php:19]: done

File: database/migrations/2020_10_29_160742_create_books_table.php (hunk 7 above)
Comment (alamriku) [reply database/migrations/2020_10_29_160742_create_books_table.php:20]: done

File: database/migrations/2020_10_29_160742_create_books_table.php (hunk 9 above)
Comment (alamriku) [reply database/migrations/2020_10_29_160742_create_books_table.php:24]: done

File: database/migrations/2020_10_29_163947_create_book_copies_table.php (hunk 11 above)
Comment (alamriku) [reply database/migrations/2020_10_29_163947_create_book_copies_table.php:18]: done

File: database/migrations/2020_10_29_163947_create_book_copies_table.php (hunk 12 above)
Comment (alamriku) [reply database/migrations/2020_10_29_163947_create_book_copies_table.php:24]: done

File: database/migrations/2020_10_29_170920_create_loan_requests_table.php (hunk 14 above)
Comment (alamriku) [reply database/migrations/2020_10_29_170920_create_loan_requests_table.php:20]: done

File: database/migrations/2020_10_29_171427_create_return_requests_table.php (hunk 15 above)
Comment (alamriku) [reply database/migrations/2020_10_29_171427_create_return_requests_table.php:20]: done

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 2 above)
Comment (alamriku) [reply database/migrations/2014_10_12_000000_create_users_table.php:23]: Don't no how to use it. google it but not found.

File: database/migrations/2020_10_29_160742_create_books_table.php (hunk 10 above)
Comment (alamriku) [reply database/migrations/2020_10_29_160742_create_books_table.php:38]: done

**Pull request merged**